prediction-applier/src/main/java/org/alfresco/hxi_connector/prediction_applier/config/AppConfig.java

```diff
@@ -0,0 +1,62 @@
+/*-
+ * #%L
+ * Alfresco HX Insight Connector
+ * %%
+ * Copyright (C) 2023 - 2024 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.hxi_connector.prediction_applier.config;
+
+import java.util.Optional;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import org.alfresco.hxi_connector.common.adapters.auth.AuthService;
+import org.alfresco.hxi_connector.common.adapters.messaging.repository.ApplicationInfoProvider;
+import org.alfresco.hxi_connector.common.adapters.messaging.repository.api.DiscoveryApiClient;
+import org.alfresco.hxi_connector.common.config.properties.Application;
+
+@Configuration
+public class AppConfig
+{
+
+    @Bean
+    @ConfigurationProperties(prefix = "application")
+    public Application application()
+    {
+        return new Application();
+    }
+
+    @Bean
+    public DiscoveryApiClient discoveryApiClient(RepositoryApiProperties repositoryApiProperties, AuthService authService, ObjectMapper objectMapper)
+    {
+        return new DiscoveryApiClient(repositoryApiProperties.discoveryEndpoint(), authService, objectMapper);
+    }
+
+    @Bean
+    public ApplicationInfoProvider applicationInfoProvider(DiscoveryApiClient discoveryApiClient, Application applicationProperties)
+    {
+        return new ApplicationInfoProvider(discoveryApiClient, applicationProperties, Optional.empty());
```

Comment:
Same here, the `versionOverride` doesn't seem to be taken into consideration 🤔

Reply:
See my previous comment.
BTW, in the next PR (adding user agent header to `hxinsight-extension` module) I am preparing a small refactoring with an abstraction layer put over getting repo version as all of the modules have different approaches/needs to getting the repo version.
I plan to create different beans conditionally on properties in `live-ingester` module, a bean basing on ACS Discovery API in `prediction-applier` module and a bean in `hxinsight-extension` module using a Java API.